### PR TITLE
Mercury pubnet

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
What
Refactors in order to allow for a Mercury URL per network
Adds pubnet to supported Mercury networks
Moves urql client construction in order to support a client per network
Updates schema for Mercury changes
Adds missing account object data to calc xlm available balance

Why
Mercury now supports pubnet

### CODE REVIEW

File: src/service/mercury/helpers/transformers.ts [hunk 1]
```diff
@@ -119,10 +119,7 @@ const transformAccountBalances = async (
   }, {} as NonNullable<AccountBalancesInterface["balances"]>);
 
   const classicBalances = classicBalanceData.reduce((prev, curr) => {
-    const codeAscii = Buffer.from(
-      curr.assetByAsset.code.substring(2),
-      "hex"
-    ).toString("utf8");
+    const codeAscii = atob(curr.assetByAsset.code);
```
Comment (aristidesstaffieri) [inline src/service/mercury/helpers/transformers.ts:122]: Encoding changed in Mercury for this.

File: src/service/mercury/queries.ts [hunk 2]
```diff
@@ -332,23 +335,23 @@ export const query = {
             code
             issuer
           }
-          assetByPath1Asset {
```
Comment (aristidesstaffieri) [inline src/service/mercury/queries.ts:335]: All of these are unrelated schema changes that Mercury introduced.

File: src/service/mercury/helpers/transformers.ts [hunk 3]
```diff
@@ -75,21 +78,21 @@ const transformAccountBalances = async (
   const classicBalanceData = rawResponse?.data?.balanceByPublicKey.nodes || [];
 
   const accountObject = accountObjectData[0];
-  // TODO: get these into query
-  const numSponsoring = 0;
-  const numSponsored = 0;
-  const sellingLiabilities = 0;
+  const numSubEntries = accountObject.numSubEntries || "0";
```
Comment (aristidesstaffieri) [inline src/service/mercury/helpers/transformers.ts:81]: unrelated, Mercury has now added these fields to the account object so added them here.

File: src/index.ts [hunk 4]
```diff
@@ -45,23 +47,51 @@ async function main() {
...
+    TESTNET: conf.mercuryGraphQLTestnet,
+    PUBLIC: conf.mercuryGraphQLPubnet,
+  };
+
+  const backends = {
+    TESTNET: conf.mercuryBackendTestnet,
+    PUBLIC: conf.mercuryBackendPubnet,
+  };
+
+  function renewClientMaker(network: NetworkNames) {
```
Comment (piyalbasu) [inline src/index.ts:60]: super NIT style thing: is the reason for using `function` rather than `const renewClientMaker = () =>`  here to hoist? If not, would probably better to follow the coding conventions in the rest of the repo

File: src/index.ts [hunk 5]
```diff
@@ -45,23 +47,51 @@ async function main() {
...
+    if (!hasIndexerSupport(network)) {
+      throw new Error(`network not currently supported: ${network}`);
```
Comment (piyalbasu) [inline src/index.ts:62]: you have similar errors in some different areas. Would it be better to vary these error messages so we can more easily trace back logs to where they were thrown?

**Review by piyalbasu — APPROVED** [review]

File: src/index.ts (hunk 4 above)
Comment (aristidesstaffieri) [reply src/index.ts]: oh that's just oversight, no hoisting needed. Fixed in 1ad7aba92e0c854149bfad7bc6c9ded2332cae61

File: src/index.ts (hunk 5 above)
Comment (aristidesstaffieri) [reply src/index.ts:62]: Would a line number give you enough context for this? I think we could give them each a different message for that but probably cleaner to align error messages to error types and use stack trace or line number to know where it threw?

Comment (piyalbasu) [reply src/index.ts:62]: Yup, I think line number is fine! Just something to trace back
